src/claimtrie.cpp

```diff
@@ -2574,10 +2562,125 @@ uint256 CClaimTrieCache::getLeafHashForProof(const std::string& currentPosition,
+                    if (itClaimRow->first == name && itClaimRow->second.outPoint == itClaimsForName->outPoint) {
+                        claims.push_back(itClaimRow->second);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    queueNameType::const_iterator itSupportQueueNameCache = getSupportQueueCacheNameRow(name, false);
+    if (itSupportQueueNameCache != supportQueueNameCache.end()) {
+        const queueNameRowType& namedSupportRow = itSupportQueueNameCache->second;
+        for (queueNameRowType::const_iterator itSupportsForName = namedSupportRow.begin(); itSupportsForName != namedSupportRow.end(); ++itSupportsForName) {
+            supportQueueType::const_iterator itSupportQueueCache = getSupportQueueCacheRow(itSupportsForName->nHeight, false);
+            if (itSupportQueueCache != supportQueueCache.end()) {
+                const supportQueueRowType& supportRow = itSupportQueueCache->second;
+                for (supportQueueRowType::const_iterator itSupportRow = supportRow.begin(); itSupportRow != supportRow.end(); ++itSupportRow) {
+                    if (itSupportRow->first == name && itSupportRow->second.outPoint == itSupportsForName->outPoint) {
+                        supports.push_back(itSupportRow->second);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    return claimsForNameType(claims, supports, nLastTakeoverHeight);
+}
+
+CAmount CClaimTrieCache::getEffectiveAmountForClaim(const std::string& name, const uint160& claimId, std::vector<CSupportValue>* supports) const
+{
+    return getEffectiveAmountForClaim(getClaimsForName(name), claimId, supports);
+}
+
+CAmount CClaimTrieCache::getEffectiveAmountForClaim(const claimsForNameType& claims, const uint160& claimId, std::vector<CSupportValue>* supports) const
+{
+    CAmount effectiveAmount = 0;
+    for (std::vector<CClaimValue>::const_iterator it = claims.claims.begin(); it != claims.claims.end(); ++it) {
+        if (it->claimId == claimId && it->nValidAtHeight < nCurrentHeight) {
+            effectiveAmount += it->nAmount;
+            for (std::vector<CSupportValue>::const_iterator it = claims.supports.begin(); it != claims.supports.end(); ++it) {
+                if (it->supportedClaimId == claimId && it->nValidAtHeight < nCurrentHeight) {
+                    effectiveAmount += it->nAmount;
+                    if (supports) supports->push_back(*it);
+                }
+            }
+            break;
+        }
+    }
+    return effectiveAmount;
+}
+
+bool CClaimTrieCache::getInfoForName(const std::string& name, CClaimValue& claim) const
+{
+    if (dirty())
+        getMerkleHash();
+
+    CClaimTrieNode* current = &(base->root);
+    nodeCacheType::const_iterator cachedNode;
+
+    for (std::string::const_iterator itName = name.begin(); current; ++itName) {
```

Comment:
use `getNodeForName` here?

Reply:
I've thinking about that, but getNodeForName fallback to base (CClaimTrie) when it's not found in cache.